agent/agent.go

```diff
@@ -183,6 +183,13 @@ LOOP:
 			grip.Info("agent loop canceled")
 			return nil
 		case <-timer.C:
+			// Check the cedar GRPC connection so we can fail early
+			// and avoid task system failures.
+			_, err := a.comm.GetCedarGRPCConn(ctx)
+			if err != nil {
+				return errors.Wrap(err, "connecting to cedar GRPC service")
```

Comment:
Should this try to check into Cedar again if it's down temporarily? Otherwise, the agent will just shut down and the host will effectively be wasted.

Reply:
The connection has built-in retries, we currently set it to 10.

Reply:
Instead of returning an error and exiting, can this just reset the sleep timer and try again later?

Reply:
Sure!